### PR TITLE
Add MuchRails::Date/Time

### PR DESCRIPTION
**Pull request by nweathersbee**
This adds `MuchRails::Date` and `MuchRails::Time` to handle
`Date` and `Time` conversions.

### CODE REVIEW

File: lib/much-rails/date.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  # @example
+  #   MuchRails::Date(nil) # => nil
+  #   MuchRails::Date(" ") # => nil
+  #   MuchRails::Date(Time.zone.today) # => Date
+  #   MuchRails::Date(Time.current) # => Date
+  #   MuchRails::Date(DateTime.current) # => Date
+  #   MuchRails::Date("07/04/2020") # => Date
+  #   MuchRails::Date("2020.07.04") # => Date
+  #   MuchRails::Date("2020-07-04T08:15:00Z") # => Date
+  def self.Date(value)
```
Comment (kellyredding) [inline lib/much-rails/date.rb]: can we switch to this `MuchRails::Date.for(value)` instead? Sorry forgot to mention this earlier.

File: lib/much-rails/date.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# @example
+#   MuchRails::Date.for(nil) # => nil
+#   MuchRails::Date.for(" ") # => nil
+#   MuchRails::Date.for(Time.zone.today) # => Date
+#   MuchRails::Date.for(Time.current) # => Date
+#   MuchRails::Date.for(DateTime.current) # => Date
+#   MuchRails::Date.for("07/04/2020") # => Date
+#   MuchRails::Date.for("2020.07.04") # => Date
+#   MuchRails::Date.for("2020-07-04T08:15:00Z") # => Date
+module MuchRails; end
```
Comment (nweathersbee) [inline lib/much-rails/date.rb:3]: Hmm...maybe these examples should go above the `self.for` method??

**Review by kellyredding — APPROVED** [review]
@nweathersbee 🚀

File: lib/much-rails/date.rb (hunk 2 above)
Comment (kellyredding) [reply lib/much-rails/date.rb:3]: Yeah, that might be good actually. 👍🏻

**Pull request merged**